### PR TITLE
Support NMap in generate_storage_alias

### PR DESCRIPTION
**Pull request by KiChjang**
Part of #8803.

### CODE REVIEW

File: frame/support/src/lib.rs [hunk 1]
```diff
@@ -154,6 +155,18 @@ macro_rules! generate_storage_alias {
...
+			$crate::generate_storage_alias!(@GENERATE_INSTANCE_STRUCT $pallet, $name);
+			type $name = $crate::storage::types::StorageNMap<
+				[<$name Instance>],
+				(
+					$( $crate::storage::types::Key<$hasher, $key>, )+
+				),
+				$value,
+			>;
+		}
+	};
```
Comment (thiolliere) [inline frame/support/src/lib.rs:169]: can we have a test for which use this generation ?

File: frame/support/src/lib.rs [hunk 2]
```diff
@@ -154,6 +155,18 @@ macro_rules! generate_storage_alias {
...
 		}
 	};
+	($pallet:ident, $name:ident => NMap<$(($key:ty, $hasher:ty),)+ $value:ty>) => {
+		$crate::paste::paste! {
+			$crate::generate_storage_alias!(@GENERATE_INSTANCE_STRUCT $pallet, $name);
+			type $name = $crate::storage::types::StorageNMap<
+				[<$name Instance>],
+				(
+					$( $crate::storage::types::Key<$hasher, $key>, )+
+				),
```
Comment (thiolliere) [inline frame/support/src/lib.rs:165]: I think this doesn't work for when N is 1, i.e. only one key.

**Review by thiolliere — APPROVED** [review]

**Comment by thiolliere** [comment]
you probably just need to merge master to fix tests

**Review by coriolinus — APPROVED** [review]

**Comment by KiChjang** [comment]
bot merge

**Pull request merged**